snapcraft/commands/remote.py

```diff
@@ -89,6 +101,26 @@ def fill_parser(self, parser: argparse.ArgumentParser) -> None:
             help="acknowledge that uploaded code will be publicly available.",
         )
 
+    def _get_build_strategy(self) -> Optional[str]:
+        """Get the build strategy from the envvar `SNAPCRAFT_REMOTE_BUILD_STRATEGY`.
+
+        :returns: A string of the strategy or None.
+
+        :raises SnapcraftError: If the variable is set to an invalid value.
+        """
+        all_strategies = {strategy.value for strategy in _Strategies}
+
+        strategy = os.getenv(_STRATEGY_ENVVAR)
+
+        if strategy and strategy not in all_strategies:
+            raise SnapcraftError(
+                f"Unknown value {strategy!r} in environment variable "
+                f"{_STRATEGY_ENVVAR!r}. Valid values are "
+                f"{humanize_list(all_strategies, 'and')}."
+            )
+
+        return strategy
```

Comment:
How about if we return the enum value directly?

```suggestion
    def _get_build_strategy(self) -> Optional[_Strategies]:
        """Get the build strategy from the envvar `SNAPCRAFT_REMOTE_BUILD_STRATEGY`.

        :returns: The strategy or None.

        :raises SnapcraftError: If the variable is set to an invalid value.
        """
        strategy = os.getenv(_STRATEGY_ENVVAR)
        
        if not strategy:
            return None

        try:
            return _Strategies(strategy)
        except ValueError:
            valid_strategies = humanize_list((strategy.value for strategy in _Strategies), "and")
            raise SnapcraftError(
                f"Unknown value {strategy!r} in environment variable "
                f"{_STRATEGY_ENVVAR!r}. Valid values are {valid_strategies}."
            )
```

Reply:
Nice suggestion, it's more pythonic.  Applied!